technical/indicators/volatility.py

```diff
@@ -26,3 +32,30 @@ def atr_percent(dataframe, period: int = 14) -> ndarray:
 
 # NATR                 Normalized Average True Range
 # TRANGE               True Range
+
+########################################
+#
+# Choppiness index
+# 
+#  theory https://www.tradingview.com/scripts/choppinessindex/
+#  slightly adapted from https://medium.com/codex/detecting-ranging-and-trending-markets-with-choppiness-index-in-python-1942e6450b58
+
+def chopiness(dataframe, period: int = 14):
+    """
+
+    :param dataframe: Dataframe containing candle data
+    :param period: Period to use for chopiness calculation (defaults to 14)
+    :return: Float value from chopiness calculation :values 0 to +100
+    """
+
+    tr1 = pd.DataFrame(dataframe['high'] - dataframe['low']).rename(columns = {0:'tr1'})
+    tr2 = pd.DataFrame(abs(dataframe['high'] - dataframe['close'].shift(1))).rename(columns = {0:'tr2'})
+    tr3 = pd.DataFrame(abs(dataframe['low'] - dataframe['close'].shift(1))).rename(columns = {0:'tr3'})
+    frames = [tr1, tr2, tr3]
+
+    tr = pd.concat(frames, axis = 1, join = 'inner').dropna().max(axis = 1)
```

Comment:
I think this can be greatly simplified ... (with small improvements to performance, too).

in the end, you're doing 3 calculations (first 3 lines) - and simply need the max for each row between these 3 series (last line).

creating a new `Dataframe` each time seems pointless as we don't care about the intermediate result (tr1, tr2 and tr3).

Something like the below (untested!) for example

```suggestion
    tr1 = dataframe['high'] - dataframe['low']
    tr2 = abs(dataframe['high'] - dataframe['close'].shift(1))
    tr3 = abs(dataframe['low'] - dataframe['close'].shift(1))

    tr = pd.concat([tr1, tr2, tr3], axis = 1, join = 'inner').dropna().max(axis = 1)
```